kingpin/actors/packagecloud.py

```diff
@@ -0,0 +1,256 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2014 Nextdoor.com, Inc
+
+"""
+:mod:`kingpin.actors.packagecloud`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The packagecloud actor allows you to delete packages from a packagecloud repo
+or check for existance. The actor supports dry mode by looking for packages to
+delete, but leaving them intact.
+
+**Required Environment Variables**
+
+:PACKAGECLOUD_ACCOUNT:
+  packagecloud account name, i.e. https://packagecloud.io/accountname
+
+:PACKAGECLOUD_TOKEN:
+  packagecloud API Token
+"""
+
+import datetime
+import logging
+import os
+import re
+
+from tornado import gen
+
+from kingpin.actors import base
+from kingpin.actors import exceptions
+from kingpin.actors.support import api
+from kingpin.constants import REQUIRED
+
+log = logging.getLogger(__name__)
+
+__author__ = 'Charles McLaughlin <charles@nextdoor.com>'
+
+ACCOUNT = os.getenv('PACKAGECLOUD_ACCOUNT', None)
+TOKEN = os.getenv('PACKAGECLOUD_TOKEN', None)
+
+
+class PackagecloudAPI(api.RestConsumer):
+
+    _ENDPOINT = 'https://packagecloud.io/api/v1/'
+    _CONFIG = {
+        'attrs': {
+            'packages': {
+                'path': 'repos/%account%/%repo%/packages.json',
+                'http_methods': {'get': {}}
+            },
+            'delete': {
+                'path': 'repos/%account%/%repo%/%distro_version%/%filename%',
+                'http_methods': {'delete': {}}
+            },
+        },
+        'auth': {
+            'user': TOKEN,
+            'pass': ''
+        }
+    }
+
+
+class PackagecloudBase(base.HTTPBaseActor):
+
+    """Simple packagecloude Abstract Base Object"""
+
+    def __init__(self, *args, **kwargs):
+        """Check required environment variables."""
+        super(PackagecloudBase, self).__init__(*args, **kwargs)
+
+        if not ACCOUNT:
+            raise exceptions.InvalidCredentials(
+                'Missing the "PACKAGECLOUD_ACCOUNT" environment variable.')
+
+        if not TOKEN:
+            raise exceptions.InvalidCredentials(
+                'Missing the "PACKAGECLOUD_TOKEN" environment variable.')
+
+        rest_client = api.RestClient()
```

Comment:
I see no changes to the default `RestClient` for exception handling. Have you looked at the PackageCloud failure return codes and verified that the defaults will work appropriately?

https://tornado-rest-client.readthedocs.org/en/latest/basic_client.html#exception-handling-in-http-requests